Syntaxes/Diagnostics.sublime-syntax

```diff
@@ -1,49 +1,67 @@
 %YAML 1.2
 ---
+# [Subl]: https://www.sublimetext.com/docs/3/syntax.html
+# [LSP]: https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md
 hidden: true
 scope: output.lsp.diagnostics
 
+variables:
+  start_of_diag_body: ^\s+(?=\d)
+  filename_and_colon: ^(.*)(:)$
+
 contexts:
   main:
     - include: diagnostic-preamble
     - include: diagnostic-body
 
   diagnostic-preamble:
-    - match: ^(.*)(:)$
+    - match: '{{filename_and_colon}}'
       captures:
         0: meta.diagnostic.preamble.lsp
         1: string.unquoted.lsp
         2: punctuation.separator.lsp
 
   diagnostic-body:
-    - match: ^\s+(?=\d)
+    - match: '{{start_of_diag_body}}'
       push:
-        - meta_scope: meta.diagnostic.body.lsp
-        - include: pop-at-end
-        - match: (\d+)(?:(:)(\d+))?
-          captures:
-            1: constant.numeric.integer.decimal.lsp
-            2: punctuation.separator.lsp
-            3: constant.numeric.integer.decimal.lsp
-          push:
-            - include: pop-at-end
-            - match: '[^\s]+'
-              scope: comment.line.lsp
-              push:
-                - include: pop-at-end
-                - match: \berror\b
-                  scope: markup.deleted.lsp markup.error.lsp
-                  push: after-diagnostic-type-token
-                - match: \bwarning\b
-                  scope: markup.heading.lsp markup.warning.lsp
-                  push: after-diagnostic-type-token
-                - match: \binfo\b
-                  scope: markup.inserted.lsp markup.info.lsp
-                  push: after-diagnostic-type-token
-
-  after-diagnostic-type-token:
-    - include: pop-at-end
-
-  pop-at-end:
+        - ensure-diag-meta-scope
+        - expect-diag-message
+        - expect-diag-type
+        - expect-linter-type
+        - expect-line-maybe-column
+
+  ensure-diag-meta-scope:
+    - meta_scope: meta.diagnostic.body.lsp
+    - match: ""  # match the empty string
+      pop: true
+
+  expect-diag-message:
+    # Various server-specific tokens may get special treatment here in the diag message.
     - match: $
       pop: true
+
+  expect-diag-type:
+    # See: https://github.com/sublimehq/Packages/issues/1036
+    # We use old markup scopes too so that old color schemes can catch up.
+    - match: \berror\b
+      scope: markup.error.lsp markup.deleted.lsp
+      pop: true
+    - match: \bwarning\b
+      scope: markup.warning.lsp markup.heading.lsp
+      pop: true
+    - match: \binfo\b
+      scope: markup.info.lsp markup.inserted.lsp
+      pop: true
+
```

Comment:
Should `match: $ pop true` be included in each of the expect sections to ensure each one is poped of at line end even if nothing was found?